### PR TITLE
Amtrak API Addition

### PR DESCRIPTION
**Pull request by ahgoldmeer**
Created new wrapper for Amtrak API instead of using pre-built one

Created train object with all major information for trains

Currently printing all trains and information to console

### CODE REVIEW

File: train-tracker/src/AmtrakAPI.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
+//Client-Side Objects and Functions for reading/manipulating decoded data returned from server
+
+/**
+ * Represents a Station as seen by a Train
+ * found in a train's {@link Train.stations} member
+ * @constructor
+ * */
+function Station() {
+    this.stationCode = null;
+    // this.getStationName = function(){
```
Comment (ldups) [inline train-tracker/src/AmtrakAPI.js:10]: Can we remove this commented out code?

**Review by ldups — APPROVED** [review]
Working great, this is awesome! Just left a couple of nitpick comments

Also, why do we need a server for this? Would it not be possible to do it all client side?

**Comment by nicholasrehac** [comment]
> Working great, this is awesome! Just left a couple of nitpick comments
> 
> Also, why do we need a server for this? Would it not be possible to do it all client side?

Amtrak.com doesn't allow cross origin resource sharing, which means we can't make requests to it from the browser. The only way to get the api data is via a proxy server

**Review by molly-pop — APPROVED** [review]
Looks good Amitai!

**Review by tjtemple — APPROVED** [review]
good to go

**Pull request merged**